docs/faq.rst

```diff
@@ -291,6 +292,86 @@ See :ref:`encrypted_repos` for more details.
           user
           <https://security.stackexchange.com/questions/14000/environment-variable-accessibility-in-linux/14009#14009>`_.
 
+Using ``BORG_PASSCOMMAND`` with a properly permissioned file
+  Another option is to create a file with a password in it in your home
+  directory and use permissions to keep anyone else from reading it. For
+  example, first create a key::
+
+    head -c 1024 /dev/urandom | xxd -p > ~/.borg-passphrase
+    chmod 400 ~/.borg-passphrase
+
+  Then in an automated script one can put::
+
+    export BORG_PASSCOMMAND="cat ~/.borg-passphrase"
+
+  and Borg will automatically use that passphrase.
+
+Using ``BORG_PASSCOMMAND`` with MacOS Keychain
+  MacOS has a native manager for secrets (such as passphrases) which is safer
+  than just using a file as it is encrypted at rest and unlocked manually
+  (fortunately, the login keyring automatically unlocks when you login). With
+  the built-in ``security`` command, you can access it from the command line,
+  making it useful for ``BORG_PASSCOMMAND``.
+
+  First generate a passphrase and use ``security`` to save it to your login
+  (default) keychain::
+
+    security add-generic-password -D secret -U -a $USER -s borg-passphrase -w $(head -c 30 /dev/urandom | xxd -p)
+
+  In your backup script retrieve it in the ``BORG_PASSCOMMAND``::
+
+    export BORG_PASSCOMMAND="security find-generic-password -a $USER -s borg-passphrase"
+
+Using ``BORG_PASSCOMMAND`` with GNOME Keyring (Ubuntu, Debian, Fedora)
+  GNOME has a keyring daemon similar to that of MacOS. However, it requires a bit more setup
+  to get the login keyring to work. First ensure ``libpam-gnome-keyring`` is installed and
+  running on login (skip this step on Fedora, as it is installed by default)::
+
+    sudo apt install gnome-keyring libpam-gnome-keyring
+    # the following commands may also be necessary if the previous command installed new packages
+    sudo sh -c "echo session optional pam_gnome_keyring.so auto_start >> /etc/pam.d/login"
+    sudo sh -c "echo password optional pam_gnome_keyring.so >> /etc/pam.d/passwd"
+    # you may need to relogin afterwards to activate the login keyring
+
+  Then add a secret to the login keyring::
+
+    sudo apt install libsecret-tools  # Ubuntu, Debian
+    sudo yum install libsecret-tools  # Fedora, RHEL
+    head -c 30 /dev/urandom | xxd -p | secret-tool store borg-repository repo-name --label="Borg Passphrase"
```

Comment:
It's harder to write tutorials for how to do it in a GUI IMO. Also if they copy-paste these instructions they can't accidentally use the wrong folder or anything (it will put them in the defaults for `secret-tool`).